### PR TITLE
fix: simplify API Server / Job permissions

### PR DESCRIPTION
**Pull request by rangoo94**
## Pull request description 

* Simplify permissions (i.e. to avoid problems with `deletecollection` for `pods/log`)
* Fix the `{{ end }}` clauses - some roles were not created (like `watchers-rb-{{namespace}}` cluster role)

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [x] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Review by exu — APPROVED** [review]

File: charts/testkube-api/templates/role.yaml [hunk 1]
```diff
@@ -676,6 +681,15 @@ rules:
       - create
       - delete
       - deletecollection
+  - apiGroups:
+      - ""
+    resources:
+      - events
```
Comment (vsukhin) [inline charts/testkube-api/templates/role.yaml:687]: events can be created

File: charts/testkube-api/templates/role.yaml [hunk 2]
```diff
@@ -873,17 +894,26 @@ rules:
...
+      - patch
+      - update
       - list
       - create
       - delete
       - deletecollection
+  - apiGroups:
+      - ""
+    resources:
+      - events
```
Comment (vsukhin) [inline charts/testkube-api/templates/role.yaml:911]: events can be created

File: charts/testkube-api/templates/role.yaml (hunk 1 above)
Comment (rangoo94) [reply charts/testkube-api/templates/role.yaml:687]: Oh, I wasn't aware that we are creating events, thanks, will add 👍

Comment (rangoo94) [reply charts/testkube-api/templates/role.yaml:687]: Actually, should it be there? Will the jobs create events? Apparently there is `events-role-{namespace}` assigned to `testkube-api-server` itself

Comment (rangoo94) [reply charts/testkube-api/templates/role.yaml:687]: Ok, so basically as in this PR 👍

**Review by dejanzele — APPROVED** [review]

**Pull request merged**